updater/fetchers/node/node.go

```diff
@@ -0,0 +1,212 @@
+// (https://nodesecurity.io).
+type NodeFetcher struct{}
+
+func init() {
+	updater.RegisterFetcher("node", &NodeFetcher{})
+}
+
+// FetchUpdate fetches vulnerability updates from the Node Security Tracker.
+func (fetcher *NodeFetcher) FetchUpdate(datastore database.Datastore) (resp updater.FetcherResponse, err error) {
+	log.Info("fetching Node vulnerabilities")
+
+	// Download JSON.
+	r, err := http.Get(url)
+	if err != nil {
+		log.Errorf("could not download Node's update: %s", err)
+		return resp, cerrors.ErrCouldNotDownload
+	}
+
+	// Defer the addition of flag information to the response.
+	defer func() {
+		resp.FlagName = updaterFlag
+	}()
+
+	//	latestUpdate, err := datastore.GetKeyValue(updaterFlag)
+	latestUpdate := ""
+	// Unmarshal JSON.
+	var advisories nodeAdvisories
+	err = json.NewDecoder(r.Body).Decode(&advisories)
+	if err != nil {
+		log.Errorf("could not unmarshal Node's JSON: %s", err)
+		return resp, cerrors.ErrCouldNotParse
+	}
+
+	resp.Vulnerabilities, resp.FlagValue = parseNPMAdvisories(advisories.Results, latestUpdate)
+
+	return resp, nil
+}
+
+func parseNPMAdvisories(advisories []nodeAdvisory, latestUpdate string) (vulnerabilities []database.Vulnerability, newUpdated string) {
+	mvulnerabilities := make(map[string]*database.Vulnerability)
+
+	for _, advisory := range advisories {
+		for _, vulnName := range advisory.CVES {
+			if latestUpdate >= advisory.Updated_at {
+				break
+			}
+			if advisory.Updated_at > newUpdated {
+				newUpdated = advisory.Updated_at
+			}
+			// Get or create the vulnerability.
+			vulnerability, vulnerabilityAlreadyExists := mvulnerabilities[vulnName]
+			if !vulnerabilityAlreadyExists {
+				vulnerability = &database.Vulnerability{
+					Name:        vulnName,
+					Link:        cveURLPrefix + strings.TrimLeft(vulnName, "CVE-"),
+					Severity:    types.Unknown,
+					Description: advisory.Overview,
+				}
+			}
+
+			// Set the priority of the vulnerability.
+			// In the JSON, a vulnerability has one urgency per advisory it affects.
+			// The highest urgency should be the one set.
+			urgency := scoreToSeverity(advisory.Cvss_score)
+			if urgency.Compare(vulnerability.Severity) > 0 {
+				vulnerability.Severity = urgency
+			}
+
+			// Create and add the feature version.
+			pkg := database.FeatureVersion{
+				Feature: database.Feature{
+					Name: advisory.Module_name,
+					Namespace: database.Namespace{
+						Name: "node:" + defaultNPMVersion,
+					},
+				},
+			}
+			if version, err := getAdvisoryVersion(advisory.Patched_versions); err == nil {
+				pkg.Version = version
+			}
+			vulnerability.FixedIn = append(vulnerability.FixedIn, pkg)
+
+			// Store the vulnerability.
+			mvulnerabilities[vulnName] = vulnerability
+		}
+	}
+
+	// Convert the vulnerabilities map to a slice
+	for _, v := range mvulnerabilities {
+		vulnerabilities = append(vulnerabilities, *v)
+	}
+
+	return
+}
+
+func scoreToSeverity(score float32) types.Priority {
```

Comment:
If this is a generic function from NVD score to a priority, we should probably move this into the types package.